architecture/additional_concepts/authentication.adoc

```diff
@@ -206,13 +206,136 @@ When using a service account as an OAuth client:
+`serviceaccounts.openshift.io/oauth-redirectreference.` such as:
+
+----
+'serviceaccounts.openshift.io/oauth-redirecturi._<name>_'
+----
+
+In its simplest form, the annotation can be used to directly specify valid
+redirect URIs. For example:
+
+----
+"serviceaccounts.openshift.io/oauth-redirecturi.first":  "https://example.com"
+"serviceaccounts.openshift.io/oauth-redirecturi.second": "https://other.com"
+----
+
+The `first` and `second` postfixes in the above example are used to separate the
+two valid redirect URIs.
+
+In more complex configurations, static redirect URIs may not be enough. For
+example, perhaps you want all ingresses for a route to be considered valid. This
+is where dynamic redirect URIs via the
+`serviceaccounts.openshift.io/oauth-redirectreference.` prefix come into play.
+
+For example:
+
+----
+"serviceaccounts.openshift.io/oauth-redirectreference.first": "{\"kind\":\"OAuthRedirectReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Route\",\"name\":\"jenkins\"}}"
+----
+
+Since the value for this annotation contains serialized JSON data, it is easier
+to see in an expanded format:
+
+----
+
+{
+  "kind": "OAuthRedirectReference",
+  "apiVersion": "v1",
+  "reference": {
+    "kind": "Route",
+    "name": "jenkins"
+  }
+}
+
+----
+
+Now you can see that an `OAuthRedirectReference` allows us to reference the
+route named `jenkins`. Thus, all ingresses for that route will now be considered
+valid.  The full specification for an `OAuthRedirectReference` is:
+
+----
+
+{
+  "kind": "OAuthRedirectReference",
+  "apiVersion": "v1",
+  "reference": {
+    "kind": ..., <1>
+    "name": ..., <2>
+    "group": ... <3>
+  }
+}
+
+----
+
+<1> `kind` refers to the type of the object being referenced. Currently, only `route` is supported.
+<2> `name` refers to the name of the object. The object must be in the same name space as the service account.
+<3> `group` refers to the group of the object. Leave this blank, as the group for a route is the empty string.
+
+Both annotation prefixes can be combined to override the data provided by the
+reference object. For example:
+
+----
+"serviceaccounts.openshift.io/oauth-redirecturi.first":  "custompath"
+"serviceaccounts.openshift.io/oauth-redirectreference.first": "{\"kind\":\"OAuthRedirectReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Route\",\"name\":\"jenkins\"}}"
+----
+
+The `first` postfix is used to tie the annotations together. Assuming that the
+`jenkins` route had an ingress of *_\https://example.com_*, now
+*_\https://example.com/custompath_* is considered valid, but
+*_\https://example.com_* is not.  The format for partially supplying override
+data is as follows:
+
+[cols="4a,8a",options="header"]
+|===
+|Type | Syntax
+|Scheme| "https://"
+|Hostname| "//website.com"
+|Port| "//:8000"
+|Path| "examplepath"
+|===
+
+[NOTE]
+====
+Not specifying a host name override will replace the host name data from the
```

Comment:
This is backwards.  If you specify a host name override then you probably made a mistake.